crates/prune/prune/src/pruner.rs

```diff
@@ -52,13 +51,38 @@ pub struct Pruner<DB> {
     event_sender: EventSender<PrunerEvent>,
 }
 
-impl<DB: Database> Pruner<DB> {
-    /// Creates a new [Pruner].
+impl<DB> Pruner<(), DB> {
```

Comment:
these should be flipped imo

Reply:
ah, that's what you meant by "flip the generics" 😄 fixed